project_generator/tools/iar.py

```diff
@@ -135,9 +135,17 @@ def _ewp_set_toolchain(self, ewp_dic, toolchain):
     def _ewp_set_name(self, ewp_dic, name):
         ewp_dic['project']['configuration']['name'] = name
 
-    def _eww_set_path(self, eww_dic, name):
+    def _eww_set_path_single_project(self, eww_dic, name):
         eww_dic['workspace']['project']['path'] = join('$WS_DIR$', name + '.ewp')
 
+    def _eww_set_path_multiple_project(self, eww_dic, projects):
+        eww_dic['workspace']['project'] = []
+        for project in self.workspace['projects']:
+            # This has an assumption all projects inside workspace
```

Comment:
@sarahmarshy I received an email with your comments to this one. 

Yes this should be handlened in the generic layer. I did something similar for uvision, so we see the patern here and just should fix it in the ProjectWorkspace() class.